src/integration_tests/test_with_instance.py

```diff
@@ -0,0 +1,128 @@
+
+        # debugging only, where we keep an instance up between processes
+        self.state, self.statefile = {}, '/tmp/.open-test-instances.txt'
+        if self.reuse_existing_stack and os.path.exists(self.statefile):
+            self.state = json.load(open(self.statefile, 'r'))
+            self.environment = self.state.get('environment', self.environment)
+
+        self.state['environment'] = self.environment # will be saved later
+
+        with settings(abort_on_prompts=True):
+            project = 'dummy1'
+            self.stackname = '%s--%s' % (project, self.environment)
+            self.stacknames.append(self.stackname)
+
+            if self.cleanup:
+                cfn.ensure_destroyed(self.stackname)
+
+            self.context, self.cfn_template, _ = cfngen.generate_stack(project, stackname=self.stackname)
+            self.region = self.context['project']['aws']['region']
+            bootstrap.create_stack(self.stackname)
+
+            lifecycle.start(self.stackname)
+
+    @classmethod
+    def tearDownClass(self): # cls, not self
+        print('tearing down')
+        super(One, self).tearDownClass()
+        if self.reuse_existing_stack:
+            print('dumping state')
+            json.dump(self.state, open(self.statefile, 'w'))
+        if self.cleanup:
+            print('cleaning up')
+            for stackname in self.stacknames:
+                cfn.ensure_destroyed(stackname)
+        print('removing temp dir')
+        # self.rm_temp_dir()
+        # self.assertFalse(os.path.exists(self.temp_dir), "failed to delete path %r in tearDown" % self.temp_dir)
+
+    def test_bootstrap_idempotence(self):
+        "the same stack cannot be created multiple times"
+        bootstrap.create_stack(self.stackname)
+
+    def test_bootstrap_wait_until_in_progress(self):
+        bootstrap._wait_until_in_progress(self.stackname)
+        bootstrap.setup_ec2(self.stackname, self.context)
+
+    def test_bootstrap_update_template_no_updates(self):
+        "a template with no changes can be updated with no problems"
+        bootstrap.update_template(self.stackname, json.load(open(self.cfn_template, 'r')))
+
+    def test_bootstrap_run_script(self):
+        with core.stack_conn(self.stackname, username=BOOTSTRAP_USER):
+            bootstrap.run_script('test.sh')
+
+    def test_core_find_ec2_instances(self):
+        self.assertEqual(len(core.find_ec2_instances(self.stackname)), 1) # 1 node is running
+        self.assertEqual(len(core.find_ec2_instances(self.stackname, state='stopped', allow_empty=True)), 0) # 0 nodes are stopped
+
+    def test_bootstrap_template_info(self):
+        resp = bootstrap.template_info(self.stackname)
+        expecting = {
+            'stack_name': self.stackname,
+            'stack_id': 'arn:aws:cloudformation:us-east-1:512686554592:stack/dummy1--luke-20180420025657-458624/afc10200-4459-11e8-8c52-500c2864aa35',
+            'outputs': {
+                'PublicIP1': '34.229.140.136',
```

Comment:
? this doesn't look stable between builds? But could still check the keys and a regex on the values